### PR TITLE
[Core] Missing Has and const methods in ModelPart

### PR DESCRIPTION
**Pull request by loumalouomega**
**Description**
This method is defined in the mesh, but not the ModelPart. I think it is interesting to have HasNode, HasCondition, HasElement

**Changelog**
- Missing Has methods in ModelPart
- Adding pGet constant methods
- Adding test

### CODE REVIEW

File: kratos/includes/model_part.h [hunk 1]
```diff
@@ -401,6 +401,12 @@ class KRATOS_API(KRATOS_CORE) ModelPart : public DataValueContainer, public Flag
 
     void AssignNode(NodeType::Pointer pThisNode, IndexType ThisIndex = 0);
 
+    /** Returns if the Node corresponding to it's identifier exists */
+    bool HasNode(IndexType NodeId, IndexType ThisIndex = 0)
+    {
+        return GetMesh(ThisIndex).HasNode(NodeId);
+    }
```
Comment (rubenzorrilla) [inline kratos/includes/model_part.h:408]:
```suggestion
    bool HasNode(IndexType NodeId)
    {
        return GetMesh(0).HasNode(NodeId);
    }
```

Just wondering if we should start getting rid of the mesh default argument.

**Review by rubenzorrilla — COMMENTED** [review]
Makes sense to me but shouldn't we export it to Python?

**Comment by loumalouomega** [comment]
> Makes sense to me but shouldn't we export it to Python?

Yes, there is more to come, wait

File: kratos/includes/model_part.h (hunk 1 above)
Comment (loumalouomega) [reply kratos/includes/model_part.h:408]: Not my decision...

**Comment by loumalouomega** [comment]
> > Makes sense to me but shouldn't we export it to Python?
> 
> Yes, there is more to come, wait

Test added, exposed to python and constant pGet methods added (required in #8548)

File: kratos/includes/model_part.h (hunk 1 above)
Comment (rubenzorrilla) [reply kratos/includes/model_part.h:408]: @KratosMultiphysics/technical-committee

**Review by pooyan-dadvand — APPROVED** [review]
@KratosMultiphysics/technical-committee approves this changes as they are useful

**Pull request merged**